src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -24,18 +28,26 @@
     private HttpServletResponse response;
     @Mock
     private RequestDispatcher requestDispatcher;
+    @Mock
+    private ProductDao productDao;
 
-    private ProductListPageServlet servlet = new ProductListPageServlet();
+    @InjectMocks
+    private ProductListPageServlet servlet ;
 
     @Before
     public void setup(){
+        MockitoAnnotations.initMocks(this);
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+
     }
 
     @Test
     public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
-
+        verify(request).setAttribute(eq("products"),anyList());
```

Comment:
To make it even more clear you could also create a mock list, make it being returned from dao (when calling findProducts) and assert that list is passed to setAttribute (instead of anyList()), let's better go this way

Reply:
Something like this  verify(request).setAttribute(anyString(),eq(productList)) ?

Reply:
verify(request).setAttribute(eq("products"), eq(productList))